File: Chapter 3 - Stacks and Queues/Stack.java
import java.util.EmptyStackException;

public class Stack<T> {
	private Node<T> top;

	public T pop() {
		if (top == null) throw new EmptyStackException();
		else {
			T temp = top.data;
			top = top.nxt;
			return temp;
		}
	}

	public void push(T item) {
		Node<T> node = new Node<T>(item);
		node.nxt = top;
		top = node;
	}

	public T peek() {
		if (top == null) throw new EmptyStackException();
		return top.data;
	}

	public boolean isEmpty() {
		return (top == null);
	}
}

File: Chapter 2 - Linked Lists/stack.py
from node import Node

class Stack:
	top = None
	
	def push(self, val):
		item = Node(val)
		item.nxt = self.top
		self.top = item

	def pop(self):
		if (self.top):
			item = self.top.data
			self.top = self.top.nxt
			return item
	
	def peek(self):
		if (self.top):
			return self.top.data

	def is_empty(self):
		return (not self.top)